lib/dor/services/workflow_service.rb

```diff
@@ -540,8 +547,12 @@ def workflow_resource_method(uri_string, meth = 'get', payload = '', opts = {})
 
           response.body
         end
-      end
+      rescue *workflow_service_exceptions_to_catch => e
+        response = e.response if e.respond_to? :response
+        response ||= {}
 
+        raise Dor::WorkflowException, "Failed to #{meth} #{base_url}/#{uri_string} (HTTP status #{response[:status]})"
+      end
```

Comment:
If a larger view is needed, Ruby 2.1+ has a built-in mechanism for this using Exception#cause, which will return the original exception in full.